Fix test msvc-abi-return-indirect-trivial-record.swift for windows aarch64

hjyamauchi opened the pull request:
The test msvc-abi-return-indirect-trivial-record.swift fails on windows aarch64 because the type of the third argument of @"??YLoadableIntWrapper@@QEAA?AU0@U0@@Z"
(@LoadableIntWrapper::operator+=(LoadableIntWrapper)) is i64 which differs from i32 for x86_64. Accommodate this as this is how Clang generates code. See https://godbolt.org/z/G76aMr5M9. Split the test per architecture.

hjyamauchi commented:
@swift-ci Please test

compnerd commented:
Can we not use a conditional on the architecture for the check rather than duplicating the entire test?

hjyamauchi committed:
Fix test msvc-abi-return-indirect-trivial-record.swift for windows aarch64

The test msvc-abi-return-indirect-trivial-record.swift fails on
windows aarch64 because the type of the third argument of
@"??YLoadableIntWrapper@@qeaa?AU0@U0@@z"
(@LoadableIntWrapper::operator+=(LoadableIntWrapper)) is i64 which
differs from i32 for x86_64. Accommodate this as this is how Clang
generates code. See https://godbolt.org/z/G76aMr5M9.

hjyamauchi commented:
> Can we not use a conditional on the architecture for the check rather than duplicating the entire test? (i.e. `CHECK-x86_64` and `CHECK-aarch64`)

Done. PTAL

compnerd reviewed (approved):
Thanks!

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test Linux platform

compnerd commented:
@swift-ci please test macOS platform
